tests/optimize/test_optimize_shot_adaptive.py

```diff
@@ -262,6 +262,87 @@ def test_single_array_argument_step(self, cost_fn, mocker, monkeypatch):
         assert np.allclose(grad[0][0, 1], np.mean(single_shot_grads[0][:5, 0, 1]))
         assert np.allclose(grad_variance[0][0, 1], np.var(single_shot_grads[0][:5, 0, 1], ddof=1))
 
+    dev = qml.device("default.qubit", wires=2, shots=100)
+
+    def ansatz(x, **kwargs):
+        qml.StronglyEntanglingLayers(x, wires=[0, 1])
+
+    expval_cost = catch_warn_ExpvalCost(ansatz, H, dev)
+    qnode = expval_cost.qnodes[0]
+
+    @pytest.mark.parametrize("cost_fn", [qnode, expval_cost])
+    def test_padded_single_array_argument_step(self, cost_fn, mocker, monkeypatch):
+        """Test that a simple QNode with a single array argument with extra dimensions correctly
+        performs an optimization step, and that the single-shot gradients generated have the
+        correct shape"""
+        opt = qml.ShotAdaptiveOptimizer(min_shots=10)
+        spy_single_shot_expval = mocker.spy(opt, "_single_shot_expval_gradients")
+        spy_single_shot_qnodes = mocker.spy(opt, "_single_shot_qnode_gradients")
+        spy_grad = mocker.spy(opt, "compute_grad")
+
+        shape = qml.StronglyEntanglingLayers.shape(n_layers=1, n_wires=2)
+        x_init = np.ones(shape) * 0.5
+        new_x = opt.step(cost_fn, x_init)

```

Comment:
When you run a second optimization step, it will fail. In the first one only the shape of opt.s is wrong. This will only collide with the parameter shape in the second optimization step.
```suggestion
        new_x = opt.step(cost_fn, x_init)
        new_x = opt.step(cost_fn, new_x)
```